File: CreditCalculator.Core/CalculationResult.cs
namespace CreditCalculator.Core;

public record CalculationResult(
    double? Payment,
    double SumPayment,
    double SumMainDebtPayment,
    double SumPercentPayment,
    double MainDebtInPercent,
    double PercentsInPercent,
    IReadOnlyCollection<PaymentInfo> PaymentInfos
);

File: credit-calculator-ui/src/models/calculation-result.ts
import { PaymentInfo } from ".";

type CalculationResult = {
    payment?: number;
    sumPayment: number;
    sumMainDebtPayment: number;
    sumPercentPayment: number;
    mainDebtInPercent: number;
    percentsInPercent: number;
    paymentInfos: PaymentInfo[];
}

export default CalculationResult;